cloudinit/util.py

```diff
@@ -397,9 +397,10 @@ def translate_bool(val, addons=None):
 
 
 def rand_str(strlen=32, select_from=None):
+    r = random.SystemRandom()
```

Comment:
[The `SystemRandom` docs](https://docs.python.org/3/library/random.html#random.SystemRandom) say that it is "Not available on all systems." Specifically, it is not available on systems where `os.urandom` would raise `NotImplementedError`:

> On a Unix-like system, random bytes are read from the /dev/urandom device. If the /dev/urandom device is not available or not readable, the NotImplementedError exception is raised.

cloud-init is used on BSD systems, so we cannot assume that we are running on top of the Linux kernel. I don't know the state of urandom on BSDs, but I would like to know that we won't be breaking behaviour there.